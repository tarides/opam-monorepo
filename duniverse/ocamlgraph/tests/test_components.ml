(**************************************************************************)
(*                                                                        *)
(*  Ocamlgraph: a generic graph library for OCaml                         *)
(*  Copyright (C) 2004-2007                                               *)
(*  Sylvain Conchon, Jean-Christophe Filliatre and Julien Signoles        *)
(*                                                                        *)
(*  This software is free software; you can redistribute it and/or        *)
(*  modify it under the terms of the GNU Library General Public           *)
(*  License version 2, with the special exception on linking              *)
(*  described in file LICENSE.                                            *)
(*                                                                        *)
(*  This software is distributed in the hope that it will be useful,      *)
(*  but WITHOUT ANY WARRANTY; without even the implied warranty of        *)
(*  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.                  *)
(*                                                                        *)
(**************************************************************************)

open Format
open Graph

module C = Components.Undirected(Pack.Graph)

open Pack.Graph

let () =
  let g = Rand.graph ~v:10 ~e:3 () in
  let n, f = C.components g in
  printf "%d components@." n;
  iter_vertex (fun v -> printf "%d -> %d@." (V.label v) (f v)) g


(*
Local Variables:
compile-command: "ocaml -I .. graph.cma test_components.ml"
End:
*)
