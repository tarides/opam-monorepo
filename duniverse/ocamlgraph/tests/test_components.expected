7 components
0 -> 0
1 -> 1
2 -> 1
3 -> 2
4 -> 3
5 -> 3
6 -> 4
7 -> 5
8 -> 6
9 -> 0
